src/SqlPersistence/Config/SqlPersistenceConfig_TablePrefix.cs

```diff
@@ -20,14 +20,17 @@ public static void TablePrefix(this PersistenceExtensions<SqlPersistence> config
 
         internal static string GetTablePrefix(this IReadOnlySettings settings)
         {
-            if (settings.TryGet("SqlPersistence.TablePrefix", out string tablePrefix))
+            var processorEndpoint = settings.GetOrDefault<string>(SqlOutboxFeature.ProcessorEndpointKey);
+
+            if (!string.IsNullOrWhiteSpace(processorEndpoint))
             {
-                return tablePrefix;
+                return TableNameCleaner.Clean(processorEndpoint);
             }
+
             var endpointName = settings.EndpointName();
-            var clean = TableNameCleaner.Clean(endpointName);
-            return $"{clean}_";
+            return settings.GetOverridenTablePrefix() ?? TableNameCleaner.Clean(endpointName);
```

Comment:
I think that if a customer has overridedn the table prefix explicitly we should honor that so I propose we check that first and only use our override if its not specified?